ssv/partial_sig_container.go

```diff
@@ -43,10 +29,43 @@ func (ps *PartialSigContainer) ReconstructSignature(root [32]byte, validatorPubK
 	return signature.Serialize(), nil
 }
 
-func (ps *PartialSigContainer) HasQuorum(root [32]byte) bool {
-	return uint64(len(ps.Signatures[rootHex(root)])) >= ps.Quorum
+// SignatureForRoot returns a map of signer and signature for a specific root
+func (ps PartialSignatureContainer) SignatureForRoot(root [32]byte) map[types.OperatorID][]byte {
+	sigs := make(map[types.OperatorID][]byte, 0)
+	for operatorID, sigMsg := range ps {
+		for _, msg := range sigMsg.Message.Messages {
+			if bytes.Equal(msg.SigningRoot[:], root[:]) {
+				sigs[operatorID] = msg.PartialSignature
+			}
+		}
+	}
+	return sigs
+}
+
+// Roots returns roots for the partial sigs
+func (ps PartialSignatureContainer) Roots() [][32]byte {
+	if len(ps) > 0 {
+		ret := make([][32]byte, 0)
+		for _, sigMsg := range ps {
+			for _, msg := range sigMsg.Message.Messages {
+				ret = append(ret, msg.SigningRoot)
+			}
+			break // only need to iterate first msg
+		}
+		return ret
+	}
```

Comment:
we could do 
```golang
if len(ps) =< 0 {
return nil // or [][32]byte{}
}
```
 to reduce nesting and add readability

Reply:
done